docs/about/contributing/modules.md

```diff
@@ -0,0 +1,406 @@
+- Providing access to running services
+
+**Key properties**:
+
+- `agent_id`: The Coder agent
+- `external`: `true` for protocol handlers, `false` for web apps
+- `group`: Name of a group that this app belongs to.
+- `healthcheck`: Monitor service availability
+- `order`: Control the display order of apps in the dashboard
+- `subdomain`: Access method for web apps
+- `url`: Service URL or protocol handler
+
+Example:
+
+```tf
+resource "coder_app" "service" {
+  agent_id = var.agent_id
+  slug     = "service"
+  url      = "http://localhost:${var.port}"
+
+  healthcheck {
+    url       = "http://localhost:${var.port}/api/status"  # Service-specific endpoint
+    interval  = 5
+    threshold = 6
+  }
+}
+```
+
+> **⚠️ Important**: Health check URLs are service-specific. Common paths include `/health`, `/healthz`, `/ping`, `/status`, or `/api/health`. Check your service's documentation or use the main service URL if no dedicated health endpoint exists.
+
+#### coder_env - Environment Variables
+
+[Set environment variables in workspace sessions](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/env) for tool configuration and authentication.
+
+**When to use**:
+
+- Configuring development tools and CLIs
+- Providing authentication tokens
+- Setting service endpoints
+
+Example:
+
+```tf
+resource "coder_env" "config" {
+  agent_id = var.agent_id
+  name     = "TOOL_CONFIG"
+  value    = jsonencode(var.config)
+}
+```
+
+#### Data Sources
+
+Access workspace context through Coder data sources:
+
+```tf
+data "coder_provisioner" "me" {}
+# Provides: arch, os for platform-specific logic
+
+data "coder_workspace" "me" {}
+# Provides: name, id, access_url, start_count
+
+data "coder_workspace_owner" "me" {}
+# Provides: name, email, full_name, groups of the workspace owner.
+```
+
+### Variable Design
+
+**Required Variables**:
+
+```tf
+variable "agent_id" {
+  type        = string
+  description = "The ID of a Coder agent."
+  # No default - always required
+}
+```
+
+**Optional Variables with Validation**:
+
+```tf
+variable "version" {
+  type        = string
+  description = "Software version to install."
+  default     = "latest"
+  validation {
+    condition     = can(regex("^(latest|[0-9]+\\.[0-9]+\\.[0-9]+)$", var.version))
+    error_message = "Version must be 'latest' or semantic version (e.g., 1.2.3)."
+  }
+}
+```
+
+**Complex Configuration Objects**:
+
+```tf
+variable "service_config" {
+  type = object({
+    port         = optional(number, 8080)
+    enable_ssl   = optional(bool, false)
+    log_level    = optional(string, "info")
+    features     = optional(list(string), [])
+  })
+  description = "Service configuration options."
+  default     = {}
+}
+```
+
+**Sensitive Variables**:
+
+```tf
+variable "api_key" {
+  type        = string
+  description = "API key for service authentication."
+  default     = ""
+  sensitive   = true
+}
+```
+
+### Script Development
```

Comment:
Same comment. I think we should drop the words Development and use an action sentence like "Running Scripts"